src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.cs

```diff
@@ -30,28 +29,20 @@ private IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document
             return formattingRules;
         }
 
-        public virtual IndentationResult? GetDesiredIndentation(Document document, int lineNumber, CancellationToken cancellationToken)
+        public IndentationResult GetDesiredIndentation(Document document, int lineNumber, CancellationToken cancellationToken)
         {
             var indenter = GetIndenter(document, lineNumber, cancellationToken);
 
             var indentStyle = indenter.OptionSet.GetOption(FormattingOptions.SmartIndent, document.Project.Language);
             if (indentStyle == FormattingOptions.IndentStyle.None)
             {
                 // If there is no indent style, then do nothing.
-                return null;
+                return new IndentationResult(basePosition: 0, offset: 0);
```

Comment:
hmmm... so this will return something from ISmartIndenter and editor think we handled it? when we don't know, we should return null to editor so it can move to next one.

Reply:
I guess fine for our own service. we can define contract whatever way we want to.

Reply:
I agree.  all that 'None' means is "we won't indent".  it's fine for us to handle in just that manner.